syncapi/routing/messages.go

```diff
@@ -70,6 +72,13 @@ func OnIncomingMessagesRequest(
 ) util.JSONResponse {
 	var err error
 
+	snapshot, err := db.NewDatabaseTransaction(req.Context())
```

Comment:
Why is this using `NewDatabaseTransaction` but /context uses `NewDatabaseSnapshot`?

Reply:
I'll add a comment about this, but the main idea is that `NewDatabaseSnapshot` is supposed to be read-only and repeatable read, whereas `NewDatabaseTransaction` isn't. Since `/messages` can backfill and store events directly in the sync API database, it needs to be able to write.